frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,188 @@
+          modal.shouldBeOpened();
+          modal.submit();
+        }
+      });
+  });
+};
+
+const dynamicNavTest = (navID: string) => {
+  nav.sidenav.clickNavLink(['Demo Plugin', `Dynamic Nav ${navID}`]);
+  cy.get('.pf-c-title.pf-m-2xl').should('contain', `Dynamic Page ${navID}`);
+  cy.get(infoMessage).should('exist');
+  cy.get(warningMessage).should('exist');
+  cy.get(hintMessage).should('exist');
+  cy.get('.pf-c-card').should('have.length', 50);
+};
+
+describe('Demo dynamic plugin test', () => {
+  before(() => {
+    cy.login();
+    nav.sidenav.switcher.changePerspectiveTo('Administrator');
+    nav.sidenav.switcher.shouldHaveText('Administrator');
+    cy.visit('/k8s/all-namespaces/import');
+    cy.window().then((win: any) => {
+      const isLocalDevEnvironment = win.SERVER_FLAGS?.authDisabled;
+      cy.readFile(`${PLUGIN_PATH}/oc-manifest.yaml`).then((textManifest) => {
+        const yamlManifest = safeLoadAll(textManifest);
+        const yamlManifestUpdatedPullSpec = yamlManifest.map((yamlDefinition) => {
+          if (yamlDefinition.kind === 'Deployment' && !isLocalDevEnvironment && PLUGIN_PULL_SPEC) {
+            return _.defaultsDeep(
+              {},
+              {
+                spec: {
+                  template: {
+                    spec: {
+                      containers: [{ name: 'console-demo-plugin', image: PLUGIN_PULL_SPEC }],
+                    },
+                  },
+                },
+              },
+              yamlDefinition,
+            );
+          }
+          return yamlDefinition;
+        });
+        const finalYAMLText = yamlManifestUpdatedPullSpec.map((obj) => safeDump(obj)).join('---\n');
+        yamlEditor.isImportLoaded();
+        yamlEditor.setEditorContent(finalYAMLText).then(() => {
+          yamlEditor.clickSaveCreateButton();
+          cy.get(errorMessage).should('not.exist');
+          cy.byTestID('success-icon').should('have.length', 5);
+          cy.byTestID('import-more-yaml').should('exist');
+          enableDemoPlugin(true);
+        });
+      });
+    });
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    enableDemoPlugin(false);
+    cy.deleteProject(PLUGIN_NAME);
+    cy.logout();
+  });
+
+  it(`test Dynamic Nav items`, () => {
+    const dynamicNavIDs = ['1', '2'];
+    dynamicNavIDs.forEach((id) => dynamicNavTest(id));
+  });
+
+  it(`test Test Consumer nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'Test Consumer']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'Extensions of type Console.flag/Model');
+    cy.get('.pf-c-card').should('have.length', 20);
+  });
+
+  it(`test Test Utilities nav item`, () => {
+    nav.sidenav.clickNavLink(['Demo Plugin', 'Test Utilities']);
+    cy.get('.pf-c-title.pf-m-2xl').should('contain', 'Utilities from Dynamic Plugin SDK');
+    cy.get('.pf-c-card__title').should('contain', 'Utility: consoleFetchJSON');
```

Comment:
It's more important to test that the utility is working. (There should be some content and no errors.)

Reply:
@spadgett tested for content and no error data